File: samples/volume-mount/VolumeMount.AppHost/apphost.ts
// Setup: Run the following commands to add required integrations:
//   aspire add sqlserver
//   aspire add azure-storage

import { createBuilder, ContainerLifetime } from "./.modules/aspire.js";

const builder = await createBuilder();

const sqlserver = await builder.addSqlServer("sqlserver")
    .withDataVolume()
    .withLifetime(ContainerLifetime.Persistent);

const sqlDatabase = sqlserver.addDatabase("sqldb");

const blobs = builder.addAzureStorage("Storage")
    .runAsEmulator(emulator => emulator.withDataVolume())
    .addBlobs("BlobConnection");

const blazorweb = builder.addProject("blazorweb")
    .withReference(sqlDatabase)
    .waitFor(sqlDatabase)
    .withReference(blobs)
    .waitFor(blobs);

await builder.build().run();

File: samples/volume-mount/cs/VolumeMount.AppHost/AppHost.cs
var builder = DistributedApplication.CreateBuilder(args);

var sqlserver = builder.AddSqlServer("sqlserver")
    .WithDataVolume()
    .WithLifetime(ContainerLifetime.Persistent);

var sqlDatabase = sqlserver.AddDatabase("sqldb");

var blobs = builder.AddAzureStorage("Storage")
    // Use the Azurite storage emulator for local development
    .RunAsEmulator(emulator => emulator.WithDataVolume())
    .AddBlobs("BlobConnection");

builder.AddProject<Projects.VolumeMount_BlazorWeb>("blazorweb")
    .WithReference(sqlDatabase)
    .WaitFor(sqlDatabase)
    .WithReference(blobs)
    .WaitFor(blobs);

builder.Build().Run();
